server/src/main/java/org/elasticsearch/indices/SystemIndices.java

```diff
@@ -643,13 +643,9 @@ private static boolean overlaps(String pattern1, String pattern2) {
      */
     public static void validateFeatureName(String name, String plugin) {
         if (SnapshotsService.NO_FEATURE_STATES_VALUE.equalsIgnoreCase(name)) {
-            throw new IllegalArgumentException(
-                "feature name cannot be reserved name [\""
-                    + SnapshotsService.NO_FEATURE_STATES_VALUE
-                    + "\"], but was for plugin ["
-                    + plugin
-                    + "]"
-            );
+            throw new IllegalArgumentException("""
+                feature name cannot be reserved name ["%s"], but was for plugin [%s]\
+                """.formatted(SnapshotsService.NO_FEATURE_STATES_VALUE, plugin));
```

Comment:
I think this would be easier more legible:
```suggestion
            throw new IllegalArgumentException(
                "feature name cannot be reserved name [\"%s\"], but was for plugin [%s]".formatted(
                    SnapshotsService.NO_FEATURE_STATES_VALUE,
                    plugin
                )
            );
```